src/Symfony/Component/ExpressionLanguage/Node/NullCoalesceNode.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ExpressionLanguage\Node;
+
+use Symfony\Component\ExpressionLanguage\Compiler;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @internal
+ */
+class NullCoalesceNode extends Node
+{
+    public function __construct(Node $expr1, Node $expr2)
+    {
+        parent::__construct(['expr1' => $expr1, 'expr2' => $expr2]);
+    }
+
+    public function compile(Compiler $compiler)
+    {
+        $compiler
+            ->raw('((')
+            ->compile($this->nodes['expr1'])
+            ->raw(') ?? (')
+            ->compile($this->nodes['expr2'])
+            ->raw('))')
+        ;
+    }
+
+    public function evaluate(array $functions, array $values)
+    {
+        set_error_handler(function ($errno, $errstr) {
+            if (\E_WARNING === $errno) {
```

Comment:
it would be really better if we could remove this error handler and rely on explicit logic instead.
this warning could come from something else, and checking the error message would look like a hack to me.
I suppose we need some cooperation between NullCoalesceNode and GetAttrNode...

Reply:
@nicolas-grekas I've just pushed a solution to this!